pyproject.toml

```diff
@@ -15,3 +15,6 @@ src_paths = ["src", "tests"]
 [build-system]
 requires = ["setuptools"]
 build-backend = "setuptools.build_meta"
+
+[tool.ruff]
+select = ["T"]
```

Comment:
In order to see if flake8 can replace flake8 and isort mid-term, let's activate all flake8 and isort features.

Reply:
*ruff

Reply:
Ah yes :P

Reply:
Turns out ruff does not use the same options with isort as isort itself. I could replicate the functionality to limit the changes to a minimum. But I can't figure out what option is triggering changes like these:
![image](https://user-images.githubusercontent.com/41837037/226685235-0ea8d001-260f-4ae6-bfca-ce4449f87bf9.png)
here are all the options ruff allows configuring - https://beta.ruff.rs/docs/settings/#isort
Also, I had to add a rule to remove line breaks between import sections too, why did we suppress them in the first place? They look better imo:
![image](https://user-images.githubusercontent.com/41837037/226685687-b3f6d18e-9d53-4120-ae82-622441f30c8e.png)

Reply:
I like them as well. But I guess @m3nu doesn't.

Reply:
> But I can't figure out what option is triggering changes like these

Maybe `PyQt5` isn't recognized as a [third party library](https://beta.ruff.rs/docs/settings/#known-third-party)?

Reply:
Either way is good. If there are many imports, some line breaks can separate the various groups of imports.

Reply:
I removed isort from the codebase and replaced it with ruff. It added some lines between imports but I think this separation is better.